docs/concepts/crds/appbinding.md

```diff
@@ -0,0 +1,133 @@
+---
+title: AppBinding Overview
+menu:
+  product_stash_0.8.3:
+    identifier: appbinding-overview
+    name: AppBinding
+    parent: crds
+    weight: 45
+product_name: stash
+menu_name: product_stash_0.8.3
+section_menu_id: concepts
+---
+
+> New to Stash? Please start [here](/docs/concepts/README.md).
+
+# AppBinding
+
+## What is AppBinding
+
+Stash needs to know how to connect with a target database and the credentials necessary to access it. This leads us to use another object called `AppBinding`.
+
+An `AppBinding` is a Kubernetes `CustomResourceDefinition (CRD)` which points to an application using either its URL (usually for a non-Kubernetes resident service instance) or a Kubernetes service object (if hosted inside a Kubernetes cluster), some optional parameters and a credential secret. To know more about why AppBinding and what problem it solves, please read this blog post: [The case for AppBinding](https://blog.byte.builders/post/the-case-for-appbinding).
+
+If you deploy a database using [KubeDB](https://kubedb.com/docs/0.11.0/concepts/), `AppBinding` object will be created automatically for it. Otherwise, you have to create an `AppBinding` object manually pointing to your desired database.
+
+## AppBinding CRD Specification
+
+Like other official Kubernetes resources, `AppBinding` has `TypeMeta`, `ObjectMeta` and `Spec` sections. However, unlike other Kubernetes resources, it does not have a `Status` section.
+
+An `AppBinding` object created by `KubeDB` for PostgreSQL database is shown below,
+
+```yaml
+apiVersion: appcatalog.appscode.com/v1alpha1
+kind: AppBinding
+metadata:
+  name: quick-postgres
+  namespace: demo
+  labels:
+    app.kubernetes.io/component: database
+    app.kubernetes.io/instance: quick-postgres
+    app.kubernetes.io/managed-by: kubedb.com
+    app.kubernetes.io/name: postgres
+    app.kubernetes.io/version: 10.2-v2
+    kubedb.com/kind: Postgres
+    kubedb.com/name: quick-postgres
+spec:
+  type: kubedb.com/postgres
+  secret:
+    name: quick-postgres-auth
+  clientConfig:
+    service:
+      name: quick-postgres
+      path: /
+      port: 5432
+      query: sslmode=disable
+      scheme: postgresql
+  secretTransforms:
+  - renameKey:
+      from: POSTGRES_USER
+      to: username
+  - renameKey:
+      from: POSTGRES_PASSWORD
+      to: password
+```
+
+Here, we are going to describe only few sections of an `AppBinding` crd that are relevant in Stash perspective.
```

Comment:
We need to document how to write AppBinding for each supported DB in the docs by hand. This will be useful for non-KubeDB managed databases. This can be done in the individual repos.